### PR TITLE
ci: Add check for gadget guides

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
Add a check to ensure gadget guides are updated and we have the necessary symlinks.

Fixes: https://github.com/inspektor-gadget/inspektor-gadget/issues/3365

### CODE REVIEW

File: gadgets/Makefile [hunk 1]
```diff
@@ -48,6 +48,13 @@ build: $(GADGETS)
 
 readmes: $(GADGETS_README)
 	counter=10 ; for i in $(shell git grep '^sidebar_position:' -- .|cut -d: -f1|grep /README.mdx) ; do sed -i "s/^sidebar_position:.*/sidebar_position: $$counter/g" $$i ; counter=$$(($$counter + 10)) ; done
+	@# Create symlinks for READMEs if not exists
+	for i in $(GADGETS_README) ; do \
+		if [ ! -f $(ROOT_DIR)../docs/gadgets/$$(dirname $$i).mdx ] ; then \
+			ln -sf ../../gadgets/$$i $(ROOT_DIR)../docs/gadgets/$$(dirname $$i).mdx ; \
```
Comment (alban) [inline gadgets/Makefile:54]: This could be moved to the `$(GADGETS_README):` target line 80, so there would not need to be a for loop.

Comment (mqasimsarfraz) [reply gadgets/Makefile]: Done.

File: gadgets/Makefile [hunk 2]
```diff
@@ -39,7 +39,7 @@ GADGETS = \
 
 DATE=$(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
 
-GADGETS_README = $(addsuffix /README.mdx, $(GADGETS))
+GADGETS_README = $(filter-out ci/sched_cls_drop/README.mdx, $(addsuffix /README.mdx, $(GADGETS)))
```
[the remark this answers is not in the document]
Comment (mqasimsarfraz) [reply gadgets/Makefile]: Good idea! Done!

File: gadgets/Makefile [hunk 3]
```diff
@@ -71,6 +71,10 @@ $(GADGETS_README):
 	@# Replace flags section
 	sed -i '/^## Flags/,/^## /{/^## /!d}' $@
 	sed -i '/## Flags/r '<(gomplate -d gadget=$(@:README.mdx=gadget.yaml) -d artifacthubpkg=$(@:README.mdx=artifacthub-pkg.yml) --file README-flags.template) $@
+	@# Create symlinks for README if not exists
+	if [ ! -f $(ROOT_DIR)../docs/gadgets/$(@) ] ; then \
+		ln -sf ../../gadgets/$@ $(ROOT_DIR)../docs/gadgets/$$(dirname $@).mdx ; \
```
[the remark this answers is not in the document]
Comment (mqasimsarfraz) [reply gadgets/Makefile]: `ROOT_DIR` already seem to be containing `/` in the end so I skipped it.

I pushed a [test commit](https://github.com/mqasimsarfraz/inspektor-gadget/commit/b68602b3dbc20ed5f46f5fc052baac74e7f8f718) on fork and seem to be working as expected: https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/10417829953/job/28853080956#step:5:206

**Comment by mqasimsarfraz** [comment]
Thanks for the reviews! The failing CI isn't related to this PR!

**Pull request merged**